objects/_switch.scss

```diff
@@ -14,21 +19,75 @@ $switch-knob-size: $switch-height - ($switch-inner-padding * 2);
     width: $switch-width;
     background: $grey-light;
     cursor: pointer;
+
     &.is-on {
         background: $blue;
     }
 }
-    .switch__knob {
-        @include border-radius($switch-height / 2);
-        @include single-transition(0.1s linear left);
+
+.switch__knob {
+    @include border-radius(50%);
+    @include single-transition(0.1s linear left);
+    position: absolute;
+    display: block;
+    left: $switch-inner-padding;
+    top: $switch-inner-padding;
+    height: $switch-knob-size;
+    width: $switch-knob-size;
+    background-color: $white;
+
+    .is-on & {
+        left: $switch-width - $switch-knob-size - $switch-inner-padding;
+    }
+}
+
+.switch--checkmark {
+    width: $switch-width-with-icon;
+
+    &::before {
```

Comment:
What does the double colon `::` do?

Reply:
this is CSS, and not SCSS, specific. `::` marks [pseudo-elements](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements)

Reply:
Ah, wait. I see you use `:` in this repo. That's deprecated for pseudo-elements:

<img width="714" alt="mdn" src="https://user-images.githubusercontent.com/985504/42173932-ed4b2820-7e20-11e8-93da-0dfc6b3501a2.png">

(MDN)

Reply:
Ah, interesting, didn't realize there was a distinction, I think we use `:` everywhere.